### PR TITLE
Add PayForMessage networking rationale

### PR DESCRIPTION
**Pull request by adlerjohn**
Add explanation for the networking logic of `PayForMessage` transaction types.

### CODE REVIEW

File: specs/proto/wire.proto [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+message TransactionFee {
+  // Max base fee rate (burnt)
+  uint64 base_rate_max = 1;
+  // Max tip fee rate (rewarded to block proposer)
+  uint64 tip_rate_max = 2;
+}
+
+message Decimal {
+  uint64 todo = 1;
```
Comment (liamsi) [inline specs/proto/wire.proto:14]: Can you link to issue #120 in a `// comment` here too?

File: specs/proto/wire.proto [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  bytes share_commitment = 2;
+  // 64-byte signature a single SignedTransactionPayForMessage
+  // https://github.com/lazyledger/lazyledger-specs/blob/master/specs/data_structures.md#signedtransactiondatapayformessage
+  bytes signature = 3;
+}
+
+message WireTxTransfer {
+  uint64 amount = 1;
+  // 32-byte address
+  bytes to = 2;
```
Comment (liamsi) [inline specs/proto/wire.proto:30]: I can hear you ranting about how protobuf doesn't have fixed size arrays :-P

File: specs/networking.md [hunk 3]
```diff
@@ -18,3 +19,13 @@ Networking
...
+
+Transaction senders who want to pay for a message will create a [SignedTransactionDataPayForMessage](./data_structures.md#signedtransactiondatapayformessage) object, `stx`, filling in the `stx.messageShareCommitment` field [based on the non-interactive default rules](../rationale/message_block_layout.md#non-interactive-default-rules) for `k = AVAILABLE_DATA_ORIGINAL_SQUARE_MAX`, then signing it to get a [transaction](./data_structures.md#transaction) `tx`. This process is repeated with successively smaller `k`s, decreasing by powers of 2 until `k * k <= stx.messageSize`. At that point, there would be insufficient shares to include both the message and transaction. Using the rest of the signed transaction data along with the pairs of `(tx.signedTransactionData.messageShareCommitment, tx.signature)`, a `WireTxPayForMessage` object is constructed.
+
+Receiving a `WireTxPayForMessage` object from the network follows the reverse process: for each `message_commitment_and_signature`, verify using the [based on the non-interactive default rules](../rationale/message_block_layout.md#non-interactive-default-rules) that the signature is valid.
```
Comment (liamsi) [inline specs/networking.md:31]: @evan-forbes from an implementer's point of view: is this understandable / sufficient?

**Review by liamsi — COMMENTED** [review]
Left a minor comment. LGTM.

File: specs/proto/wire.proto [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  uint64 message_size = 4;
+  bytes message = 5;
+  repeated MessageCommitmentAndSignature message_commitment_and_signature = 6;
+}
+
+message WireTxCreateValidator {
+  TransactionFee fee = 1;
+  uint64 nonce = 2;
+  Decimal commission_rate = 3;
+}
```
Comment (liamsi) [inline specs/proto/wire.proto:49]: https://github.com/cosmos/cosmos-sdk/blob/926ef341bfb5d64c991164f8f1571ce154368ba7/x/staking/types/tx.pb.go#L39-L48

gogp proto flavour: https://github.com/cosmos/cosmos-sdk/blob/926ef341bfb5d64c991164f8f1571ce154368ba7/proto/cosmos/staking/v1beta1/tx.proto#L36-L51

File: specs/proto/wire.proto (hunk 1 above)
Comment (adlerjohn) [reply specs/proto/wire.proto:15]: Linked in 787aa4c.

File: specs/networking.md (hunk 3 above)
Comment (evan-forbes) [reply specs/networking.md:31]: yeah, definitely. It could  be because both of you explained it to me in person, but I think it's crystal clear.

**Review by liamsi — APPROVED** [review]
🎉LGTM

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**